code/game/objects/items/devices/lightreplacer.dm

```diff
@@ -88,25 +88,26 @@
 			return
 		if(!user.unEquip(I))
 			return
-		AddUses(round(increment * 0.75))
+		AddUses(round(increment * 1)) // It literally contains the same amount of glass as a sheet.
 		to_chat(user, "<span class='notice'>You insert a shard of glass into [src]. You have [uses] light\s remaining.</span>")
 		qdel(I)
 		return
 
 	if(istype(I, /obj/item/light))
 		var/obj/item/light/L = I
-		if(L.status == 0) // LIGHT OKAY
-			if(uses < max_uses)
-				if(!user.unEquip(L))
-					return
-				AddUses(1)
-				qdel(L)
-		else
+		if(L.status == LIGHT_OK && uses < max_uses) // LIGHT OKAY
 			if(!user.unEquip(L))
 				return
-			to_chat(user, "<span class='notice'>You insert [L] into [src].</span>")
-			AddShards(1, user)
+			AddUses(1)
+			to_chat(user, "<span class='notice'>You insert [L] into [src]. You have [uses] light\s remaining.</span>")
 			qdel(L)
+			return
+
+		if(!user.unEquip(L))
```

Comment:
You refactored the `if`-statement to pass to this `if(!user.unEquip(..` if it has max uses. Will this add shards over the cap?

Reply:
It has been refactored again, you immediately return if you are at or above the limit.